src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1610,6 +1610,82 @@ bool SQLStorage::loadEcuReportCounter(std::vector<std::pair<Uptane::EcuSerial, i
   return true;
 }
 
+void SQLStorage::saveReportEvent(const Json::Value& json_value) {
+  std::string json_string = Utils::jsonToCanonicalStr(json_value);
+  SQLite3Guard db = dbConnection();
+  int64_t id = 0;
+  {
+    auto statement = db.prepareStatement("SELECT MAX(id) FROM report_events;");
+    int statement_result = statement.step();
+
+    if (statement_result != SQLITE_DONE && statement_result != SQLITE_ROW) {
+      LOG_INFO << "Can't get report_events: " << db.errmsg();
+    } else {
+      id = statement.get_result_col_int(0) + 1;
+    }
+  }
+
+  auto statement = db.prepareStatement<int64_t, std::string>(
+      "INSERT OR REPLACE INTO report_events (id, json_string) VALUES "
+      "(?,?);",
+      id, json_string);
+  if (statement.step() != SQLITE_DONE) {
+    LOG_ERROR << "Can't set ecu counter: " << db.errmsg();
+    return;
+  }
+}
+
+bool SQLStorage::loadReportEvents(Json::Value* report_array, int64_t* id_max) {
+  SQLite3Guard db = dbConnection();
+  std::vector<std::pair<Uptane::EcuSerial, int64_t>> ecu_cnt;
+  // keep the same order as in ecu_serials (start with primary)
```

Comment:
It looks like this comment was copy-pasted by mistake?

Reply:
Yes, removed.